packages/botonic-plugin-ai-agents/src/ai-agent-client.ts

```diff
@@ -1,112 +1,73 @@
-/* eslint-disable @typescript-eslint/naming-convention */
-import { BotContext, isDev } from '@botonic/core'
-import axios from 'axios'
+import { BaseChatModel } from '@langchain/core/language_models/chat_models'
+import {
+  AIMessage as LangchainAIMessage,
+  ToolMessage as LangchainToolMessage,
+} from '@langchain/core/messages'
+import { StructuredTool } from '@langchain/core/tools'
+import { CompiledStateGraph } from '@langchain/langgraph'
+import { createReactAgent } from '@langchain/langgraph/prebuilt'
 
-import { HUBTYPE_API_URL } from './constants'
-import { AiAgentError } from './errors'
+import { EXIT_TOOLS } from './tools'
 import {
+  AgenticInputMessage,
+  AgenticOutputMessage,
   AiAgentArgs,
-  AiAgentRequestData,
-  AiAgentRequestDataTest,
-  AiAgentResponse,
-  Config,
+  AssistantMessage,
+  ExitMessage,
+  ToolMessage,
 } from './types'
 
 export class AiAgentClient {
-  private url: string
+  public name: string
+  public instructions: string
+  public agent: CompiledStateGraph<any, any> // TODO: apply RunInput, RunOutput, etc.
 
-  constructor() {
-    this.url = `${HUBTYPE_API_URL}/external/v1/ai/agent`
-  }
+  constructor(
+    aiAgentArgs: AiAgentArgs,
+    chatModel: BaseChatModel,
+    tools: StructuredTool[] = []
+  ) {
+    this.name = aiAgentArgs.name
+    this.instructions = aiAgentArgs.instructions
```

Comment:
extra: maybe this instructions and name could be avoid to be saved since we do not use them

Reply:
applied in the following PR